### DISCUSSION TITLE
Usage of IDs

### DISCUSSION THREAD

**Discussion by DarkWiiPlayer** [NowhereLTD/HehIconPack#6] [General]
I already commented this on #1, but there's the question of whether IDs should be used within SVG-files.

# Advantages

IDs are the only way to use several features of SVG, such as `<use>` tags to repeat one element several times in a graphics

```svg
<defs>
   <rect id="square" width="5" height="5" color="red" />
</defs>

<use href="#square" />
<use href="#square" />
<use href="#square" />
```

or use elements from other SVG elements on the document or even external SVG files

```html
<h1>
   Follow us on twitter <svg><use href="/graphics/social.svg#twitter"/></svg>
</h1>
```

as well as linking to predefined views in an SVG document

```svg
<view id="front" viewBox="0 0 100 200" />
<view id="back" viewBox="100 0 100 200" />
```

```html
<object data="/graphics/whatever.svg#front">
<object data="/graphics/whatever.svg#front">
<!-- Note that this example is based on a real use-case I've encountered -->
```

# Disadvantages

SVG-Files can easily be embedded into HTML pages by server-side tools, in which case special care has to be taken not to embed the same SVG-File twice should it have any IDs, as these are not allowed to appear repeatedly within one element (although most browsers will not care, HTML validators will complain and thus hurt SEO).

This could, of course, be mitigated by server-side tools that take care of not repeatedly embedding one SVG and instead only embed an empty SVG with a single `<use>` tag pointing to the first appearance. This can, however, lead to unexpected CSS interactions, as the `use`d element is now taken outside of its surrounding environment in the SVG, as well as potentially causing render bugs (one such bug in chrome has only recently been fixed, where the interaction of `<use>` and CSS-@keyframe animations was causing elements to sometimes appear with the wrong colour, presumably due to faulty caching logic).

**Comment by DrMcHack** [comment]
Nice idea.

I think we determine a namespace like `heh-icon-<name of icon>-<name of id> or also to guarantee unique id's in our svg files.
